content/guides/styling-your-components.mdx

```diff
@@ -0,0 +1,522 @@
+
+```js
+import { macaronVitePlugin } from '@macaron-css/vite';
+import { defineConfig } from 'vite';
+
+export default defineConfig({
+  plugins: [
+    macaronVitePlugin(),
+    // other plugins
+  ],
+});
+```
+
+#### Usage
+
+1. Import `styled` from `@macaron-css/solid` and create a styled component:
+
+```jsx
+// button.tsx
+import { styled } from '@macaron-css/solid';
+
+const Button = styled('button', {});
+```
+
+2. Add styles that will be applied to the components by default:
+
+```jsx
+import { styled } from '@macaron-css/solid';
+
+const Button = styled('button', {
+  base: {
+    backgroundColor: 'red',
+    borderRadius: '10px',
+  },
+});
+```
+Variants can be added using the `variants` key:
+
+```jsx
+mport { styled } from '@macaron-css/solid';
+
+const Button = styled('button', {
+    base: {
+      backgroundColor: 'red',
+      borderRadius: '10px',
+    },
+    variants: {
+      color: {
+        violet: {
+          backgroundColor: 'violet',
+        },
+        gray: {
+          backgroundColor: 'gray',
+        },
+      }
+    }
+  });
+```
+
+Additionally, the `defaultVariants` feature is set to `variants` by default. This can be overrided at the time of usage:
+
+```jsx
+mport { styled } from '@macaron-css/solid';
+
+const Button = styled('button', {
+    base: {
+      backgroundColor: 'red',
+      borderRadius: '10px',
+    },
+    variants: {
+      color: {
+        violet: {
+          backgroundColor: 'violet',
+        },
+        gray: {
+          backgroundColor: 'gray',
+        },
+      }
+    },
+    defaultVariants: {
+      color: 'blue'
+    },
+  });
+```
+
+These components can be used like any other Solid component, with type-safe props derived from your variants. 
+For more information on how to use macaron, visit their [documentation](https://macaron.js.org/docs/installation).
+
+## CSS frameworks
+
+CSS frameworks provide pre-styled components and utility classes to speed up development. 
+
+### TailwindCSS
+
+[Tailwind CSS](https://tailwindcss.com/) is an on-demand utility CSS library that integrates seamlessly with Solid as a built-in PostCSS plugin.
+
+#### Installation
+1. Install Tailwind CSS as a development dependency:
+
+```bash
+npm i --save-dev tailwindcss postcss autoprefixer
+
+# or 
+
+pnpm i --save-dev tailwindcss postcss autoprefixer
+
+# or
+
+yarn add --dev tailwindcss postcss autoprefixer
+
+# Initialize
+
+npx tailwindcss init -p
+```
+2. Since TailwindCSS is configuration-driven, after initializing, a `tailwind.config.js` file will be created at the root of your project directory:
+
+```js
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+  content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx}"],
+  theme: {
+    extend: {},
+  },
+  plugins: [],
+};
+```
+
+For a deeper dive into configuration, you can check out the [Tailwind Official Documentation](https://chat.openai.com/#link-to-tailwind-docs).
```

Comment:
Did you mean for the URL to be a ChatGPT link? Or am I missing something?

Should it be this? https://v2.tailwindcss.com/docs